appinventor/blocklyeditor/src/blocks/helpers.js

```diff
@@ -122,3 +122,58 @@ Blockly.Blocks['helpers_dropdown'] = {
     return tb;
   }
 }
+
+Blockly.Blocks['helpers_screen_names'] = {
+  init: function() {
+    var utils = Blockly.Blocks.Utilities;
+    var dropdown = new Blockly.FieldInvalidDropdown(
+        this.generateOptions.bind(this));
+
+    this.setColour(Blockly.COLOUR_HELPERS);
+
+    this.setOutput(true, utils.YailTypeToBlocklyType('text', utils.OUTPUT));
+    this.appendDummyInput()
+        .appendField(dropdown, 'SCREEN');
+  },
+
+  domToMutation: function(xml) {
+    var value = xml.getAttribute('value');
+    this.setFieldValue(value, 'SCREEN');
+  },
+
+  getScreens: function() {
+    if (!this.workspace) {
+      return [];
+    }
+    var screens = Blockly.mainWorkspace.getScreenList();
+    screens.splice(screens.indexOf('Screen1'), 1);
+    console.log(screens);
+    return screens;
+  },
+
+  generateOptions: function() {
+    var screens = this.getScreens();
+    if (!screens.length) {
+      return [['', '']]
+    }
+    return screens.map(function (elem) {
+      return [elem, elem];
+    });
+  },
+
+  typeblock: function() {
+    var tb = [];
+
+    var screens = this.getScreens();
+    for (var i = 0, screen; (screen = screens[i]); i++) {
+      tb.push({
+        translatedName: Blockly.Msg.LANG_SCREENS_TITLE + screen,
```

Comment:
The options for the screen names did not appear in typeblocking when I tried it.